configure.ac

```diff
@@ -1532,6 +1532,7 @@ if test "$use_reduce_exports" = "yes"; then
   AX_CHECK_COMPILE_FLAG([-fvisibility=hidden], [CORE_CXXFLAGS="$CORE_CXXFLAGS -fvisibility=hidden"],
   [AC_MSG_ERROR([Cannot set hidden symbol visibility. Use --disable-reduce-exports.])], [$CXXFLAG_WERROR])
   AX_CHECK_LINK_FLAG([-Wl,--exclude-libs,ALL], [RELDFLAGS="-Wl,--exclude-libs,ALL"], [], [$LDFLAG_WERROR])
+  AX_CHECK_LINK_FLAG([-Wl,-no_exported_symbols], [RELDFLAGS="$RELDFLAGS -Wl,-no_exported_symbols"], [], [$LDFLAG_WERROR])
```

Comment:
This makes `libbitcoinconsensus.0.dylib` export no symbols, which is not what we want, right?
```
% dyld_info -exports src/.libs/libbitcoinconsensus.0.dylib        
src/.libs/libbitcoinconsensus.0.dylib [arm64]:
    -exports:
        offset      symbol
```

Reply:
Moved this into `LIBTOOL_APP_LDFLAGS`.